array.c

```diff
@@ -212,6 +212,33 @@ ary_embeddable_p(long capa)
 #endif
 }
 
+bool
+rb_ary_pool_moveable_p(VALUE ary)
+{
+    // if the array is shared or a shared root then it's not moveable
+    return !(ARY_SHARED_P(ary) || ARY_SHARED_ROOT_P(ary));
+}
```

Comment:
Calling this `rb_ary_pool_moveable_p` is misleading because the array is movable, just not re-embeddable.